homeassistant/components/ads/__init__.py

```diff
@@ -219,7 +237,76 @@ def add_device_notification(self, name, plc_datatype, callback):
                     "Added device notification %d for variable %s", hnotify, name
                 )
 
-    def _device_notification_callback(self, notification, name):
+    async def heartbeat(
+        self,
+        heartbeat_interval: float = 5.0,
+        max_wait_time: float = 120.0,
+        communication_timeout_while_disconnected_ms: int = 100,
+    ) -> None:
+        """Periodically checks and handles the connection state of the client.
+
+        Adjusts the wait time and communication timeout based on the connection state, with exponential backoff when disconnected.
+
+        Attempts reconnection and manages device notifications upon reconnection.
+        """
+        deleted_device_notifications: list[DeletedNotification] = []
+
+        default_timeout_ms = 5000
+
+        wait_time = heartbeat_interval
+
+        while self._is_running:
+            connection_state = self._check_connection()
+            if connection_state == ConnectionState.Connected:
+                if wait_time != heartbeat_interval:
+                    _LOGGER.info("Reconnected")
```

Comment:
Don't log on info

Reply:
This is in fact an example of where we *should* log with level info